src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double LOAD_FACTOR = 1.5;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        resizeIfNeeded();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index, size + 1);
+        resizeIfNeeded();
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index, size);
+        return (T) elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index, size);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T removedElemen = null;
+        checkIndex(index, size);
+        for (int i = 0; i < size; i++) {
+            if (i == index) {
+                removedElemen = (T) elements[i];
+            }
+        }
+        System.arraycopy(elements, index + 1, elements, index, size - index - 1);
+        size--;
+        return removedElemen;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        T removedElement = null;
+        int index = -1;
+        for (int i = 0; i < size; i++) {
+            if (elements[i] == element || elements[i] != null && elements[i].equals(element)) {
+                index = i;
+                removedElement = (T) elements[i];
+            }
+        }
+        if (index == -1) {
+            throw new NoSuchElementException("No such element was found");
+        } else {
+            System.arraycopy(elements, index + 1, elements, index, size - 1 - index);
+            size--;
+        }
```

Comment:
Can you avoid using this if-else construction?

Reply:
I think I can not avoid if else, because if index != -1 in block else i must do 3 operation that dependent of index.
Or I dont know how to avoid if-else.